src/filelock/_api.py

```diff
@@ -165,6 +174,7 @@ def acquire(
         lock_id = id(self)
         lock_filename = self._lock_file
         start_time = time.monotonic()
+        previous_umask = os.umask(0)
```

Comment:
Shouldn't this try/catch be scopped just to _acquire?

Reply:
Looking at _acquire; it will never throw a BaseException error, so it should be safe to just wrap self._acquire() with a set and reset.

The try block is probably a candidate for future improvement because it's quite strangely written; but that seems outside of the scope of this PR.